tests/integ/modin/series/test_reindex.py

```diff
@@ -300,15 +300,16 @@ def test_reindex_index_non_overlapping_index():
     )
 
 
-@sql_count_checker(query_count=2, join_count=1)
+@sql_count_checker(query_count=1, join_count=3)
 def test_reindex_index_non_overlapping_datetime_index():
+    # TODO: SNOW-1638397 See if it's possible to use data={"prices": [100, 101, np.nan, 100, 89, 88]} instead.
     date_index = native_pd.date_range("1/1/2010", periods=6, freq="D")
     native_series = native_pd.Series(
-        {"prices": [100, 101, np.nan, 100, 89, 88]}, index=date_index
+        {"1/1/2023": [100, 101, np.nan, 100, 89, 88]}, index=date_index
```

Comment:
Don't change the test case but use `xfail` for those cases.

Reply:
What's the error for those cases?

Reply:
I believe the error here is that Snowflake cannot convert "prices" to a datetime object; I'll post the exact error in a bit. But I can switch this to xfail

Reply:
The error is because in the join the original index `prices`'s type is string and then new ones are timestamps. Can you try to reuse reindex in the init constructor?